webapp/channels/src/components/admin_console/ip_filtering/index.tsx

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {AlertOutlineIcon} from '@mattermost/compass-icons/components';
+import type {AllowedIPRange, FetchIPResponse} from '@mattermost/types/config';
+
+import {Client4} from 'mattermost-redux/client';
+
+import AdminHeader from 'components/widgets/admin_console/admin_header';
+
+import IPFilteringAddOrEditModal from './add_edit_ip_filter_modal';
+import DeleteConfirmationModal from './delete_confirmation';
+import EditSection from './edit_section';
+import EnableSectionContent from './enable_section';
+import {isIPAddressInRanges} from './ip_filtering_utils';
+import SaveConfirmationModal from './save_confirmation_modal';
+
+import SaveChangesPanel from '../team_channel_settings/save_changes_panel';
+
+import './ip_filtering.scss';
+
+const IPFiltering = () => {
+    const {formatMessage} = useIntl();
+    const [showAddModal, setShowAddModal] = useState(false);
+    const [editFilter, setEditFilter] = useState<AllowedIPRange | null>(null);
+    const [ipFilters, setIpFilters] = useState<AllowedIPRange[] | null>(null);
+    const [originalIpFilters, setOriginalIpFilters] = useState<AllowedIPRange[] | null>(null);
+    const [saveNeeded, setSaveNeeded] = useState(false);
+    const [filterToDelete, setFilterToDelete] = useState<AllowedIPRange | null>(null);
+    const [currentUsersIP, setCurrentUsersIP] = useState<string | null>(null);
+    const [currentIPIsInRange, setCurrentIPIsInRange] = useState<boolean>(false);
+    const [saveConfirmationModal, setSaveConfirmationModal] = useState<JSX.Element | null>(null);
+    const [saving, setSaving] = useState<boolean>(false);
+
+    const [filterToggle, setFilterToggle] = useState<boolean>(false);
+
+    useEffect(() => {
+        Client4.getIPFilters().then((res) => {
+            setIpFilters(res as AllowedIPRange[]);
+            setOriginalIpFilters(res as AllowedIPRange[]);
+        });
+
+        Client4.getCurrentIP().then((res) => {
+            setCurrentUsersIP((res as FetchIPResponse)?.IP);
+        });
+    }, []);
+
+    useEffect(() => {
+        if (ipFilters === null || originalIpFilters === null) {
+            return;
+        }
+
+        // Check if the ipFilters list differs from the originalIpFilters list
+        const haveFiltersChanged = JSON.stringify(ipFilters) !== JSON.stringify(originalIpFilters);
+        setSaveNeeded(haveFiltersChanged);
+    }, [ipFilters, originalIpFilters]);
+
+    useEffect(() => {
+        if (!filterToggle) {
+            setCurrentIPIsInRange(true);
+            return;
+        }
+        if (!ipFilters?.length) {
+            setCurrentIPIsInRange(true);
+            return;
+        }
+        setCurrentIPIsInRange(ipFilters !== null && currentUsersIP !== null && isIPAddressInRanges(currentUsersIP, ipFilters));
```

Comment:
Do we need to store `currentIPIsInRange` in a `useState` or can we just recalculate it every time this component is updated? Generally, we don't want to store derived values in state if we can avoid it

Reply:
I've made this a derived value